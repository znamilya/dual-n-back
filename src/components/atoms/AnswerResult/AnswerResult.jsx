import React, { PropTypes } from 'react';
import bemCl from 'bem-cl';

import './AnswerResult.styl';


const b = bemCl('answer-result');

class AnswerResult extends React.Component {

    static propTypes = {
        has: PropTypes.bool,
        correct: PropTypes.bool,
    };


    /* ------------------------------------------------------------------------------------------ */
    /* METHODS                                                                                    */
    /* ------------------------------------------------------------------------------------------ */
    resolveText(isCorrect) {
        if (isCorrect) {
            return 'Correct!';
        }

        return 'Wrong';
    }


    /* ------------------------------------------------------------------------------------------ */
    /* RENDER                                                                                     */
    /* ------------------------------------------------------------------------------------------ */
    render() {
        if (!this.props.has) {
            return null;
        }

        return (
            <div className={b({
                correct: this.props.correct,
                wrong: !this.props.correct,
            })}>
                {this.resolveText(this.props.correct)}
            </div>
        );
    }
}


export default AnswerResult;
