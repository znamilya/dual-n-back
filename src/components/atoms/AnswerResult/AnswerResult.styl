.answer-result
    text-align: center


    &_correct
        color: yellowgreen


    &_wrong
        color: tomato
