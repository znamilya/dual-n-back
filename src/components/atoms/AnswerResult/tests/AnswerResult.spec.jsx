import React from 'react';

import AnswerResult from '../AnswerResult';


describe('<AnswerResult />', () => {
    it('should not be rendered if there is no an answer', () => {
        const component = shallow(
            <AnswerResult />
        );
        expect(component).toMatchSnapshot();
    });

    it('should render respective text if there is an answer', () => {
        let component = shallow(
            <AnswerResult has correct={true} />
        );
        expect(component).toMatchSnapshot();

        component = shallow(
            <AnswerResult has correct={false} />
        );
        expect(component).toMatchSnapshot();
    });
});
