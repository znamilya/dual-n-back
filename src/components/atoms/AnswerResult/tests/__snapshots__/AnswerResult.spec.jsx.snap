exports[`<AnswerResult /> should not be rendered if there is no an answer 1`] = `null`;

exports[`<AnswerResult /> should render respective text if there is an answer 1`] = `
<div
  className="answer-result answer-result_correct">
  Correct!
</div>
`;

exports[`<AnswerResult /> should render respective text if there is an answer 2`] = `
<div
  className="answer-result answer-result_wrong">
  Wrong
</div>
`;
